pkg/server/service/loadbalancer/wrr/wrr.go

```diff
@@ -151,5 +149,15 @@ func (b *Balancer) AddService(name string, handler http.Handler, weight *int) {
 	if weight != nil {
 		w = *weight
 	}
-	b.handlers = append(b.handlers, &namedHandler{Handler: handler, name: name, weight: w})
+	if w <= 0 { // non-positive weight is meaningless
+		return
+	}
+	h := &namedHandler{Handler: handler, name: name, weight: float64(w)}
+	h.deadline = b.curDeadline + 1/h.weight
```

Comment:
b.curDeadline is concurrently mutated in nextServer, so shouln't there be a read lock to protect it here?

Reply:
fixed